File: frontend/src/types/quotation.ts
export type QuotationStatus = 'Draft' | 'Submitted' | 'Selected' | 'NotSelected'

export interface QuotationItem {
  id: number
  purchaseRequestItemId: number
  productId: number
  productCode: string
  productName: string
  unitOfMeasureCode: string
  quantity: number
  unitPrice: number
  lineTotal: number
}

export interface Quotation {
  id: number
  quotationNumber: string
  purchaseRequestId: number
  purchaseRequestNumber: string
  supplierId: number
  supplierCode: string
  supplierName: string
  supplierQuotationReference: string | null
  quotationDate: string
  validUntil: string | null
  notes: string | null
  status: QuotationStatus
  totalAmount: number
  createdByUserId: number
  createdByName: string
  createdAtUtc: string
  updatedAtUtc: string | null
  submittedAtUtc: string | null
  selectedAtUtc: string | null
  items: QuotationItem[]
}

export interface QuotationItemPriceInput {
  purchaseRequestItemId: number
  unitPrice: number
}

export interface CreateQuotationRequest {
  purchaseRequestId: number
  supplierId: number
  supplierQuotationReference: string | null
  quotationDate: string
  validUntil: string | null
  notes: string | null
  items: QuotationItemPriceInput[]
}

export type UpdateQuotationRequest = Omit<
  CreateQuotationRequest,
  'purchaseRequestId' | 'supplierId'
>

export type QuotationFormValues = CreateQuotationRequest

export interface QuotationComparisonEntry {
  quotationId: number
  quotationNumber: string
  supplierId: number
  supplierCode: string
  supplierName: string
  supplierQuotationReference: string | null
  quotationDate: string
  validUntil: string | null
  status: QuotationStatus
  totalAmount: number
  isLowestTotal: boolean
  isSelected: boolean
  items: QuotationItem[]
}

export interface QuotationComparison {
  purchaseRequestId: number
  purchaseRequestNumber: string
  selectedQuotationId: number | null
  lowestTotalAmount: number | null
  quotations: QuotationComparisonEntry[]
}

File: backend/Project1.Api/Entities/Quotation.cs
namespace Project1.Api.Entities;

public sealed class Quotation
{
    public int Id { get; set; }

    public string QuotationNumber { get; set; } = string.Empty;

    public int PurchaseRequestId { get; set; }

    public PurchaseRequest PurchaseRequest { get; set; } = null!;

    public int SupplierId { get; set; }

    public Supplier Supplier { get; set; } = null!;

    public string SupplierCode { get; set; } = string.Empty;

    public string SupplierName { get; set; } = string.Empty;

    public string? SupplierQuotationReference { get; set; }

    public DateOnly QuotationDate { get; set; }

    public DateOnly? ValidUntil { get; set; }

    public string? Notes { get; set; }

    public QuotationStatus Status { get; set; } = QuotationStatus.Draft;

    public int CreatedByUserId { get; set; }

    public string CreatedByName { get; set; } = string.Empty;

    public DateTimeOffset CreatedAtUtc { get; set; } = DateTimeOffset.UtcNow;

    public DateTimeOffset? UpdatedAtUtc { get; set; }

    public DateTimeOffset? SubmittedAtUtc { get; set; }

    public DateTimeOffset? SelectedAtUtc { get; set; }

    public ICollection<QuotationItem> Items { get; set; } = [];
}
